src/main/resources/GoogleApps/Groovy.groovy

```diff
@@ -365,27 +368,32 @@ public class GoogleAppsGroovy {
             }
         }
         String id = user.get("resourceName");
-        if(id.startsWith("people/") && id.length()>7) id = id.substring(7);
+        if(id.startsWith("people/") && id.length()>7) {
+            id = id.substring(7);
+        }
         def db = context.getDatabase()
         try {
             // Force main wiki database to create the user as global
             context.setDatabase("xwiki")
-            if(email == "")
-                email = (user.emailAddresses!=null && user.emailAddresses.size()>0) ? user.emailAddresses[0].value : "";
+            if(usersEmailAddress == "") {
+                usersEmailAddress = (user.emailAddresses!=null && user.emailAddresses.size()>0) ? user.emailAddresses[0].value : "";
+            }
             def wikiUserList = services.query.xwql("from doc.object(GoogleApps.GoogleAppsAuthClass) as auth where auth.id=:id").bindValue("id", id).execute()
             if ((wikiUserList==null) || (wikiUserList.size()==0))
-                wikiUserList = services.query.xwql("from doc.object(XWiki.XWikiUsers) as user where user.email=:email").bindValue("email", email).execute()
+                wikiUserList = services.query.xwql("from doc.object(XWiki.XWikiUsers) as user where user.email=:email").bindValue("email", usersEmailAddress).execute()
 
             if ((wikiUserList==null) || (wikiUserList.size()==0)) {
```

Comment:
This test is duplicated few lines above.

Reply:
However, the code can be optimized after this fix is applied

Reply:
I suggest indeed to make these optimizations in the java part.